src/sprout/Helpers/WorkerFilesBackendMigrate.php

```diff
@@ -0,0 +1,373 @@
+<?php
+/*
+ * Copyright (C) 2017 Karmabunny Pty Ltd.
+ *
+ * This file is a part of SproutCMS.
+ *
+ * SproutCMS is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU General Public License as published by the Free Software Foundation, either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * For more information, visit <http://getsproutcms.com>.
+ */
+
+namespace Sprout\Helpers;
+
+use Kohana;
+use Sprout\Models\FileModel;
+use Sprout\Models\FileTransformModel;
+
+class WorkerFilesBackendMigrate extends WorkerBase
+{
+    protected $job_name = 'Files Backend Migration';
+
+
+    /** @var FilesBackend */
+    private $_old_backend;
+
+    /** @var FilesBackend */
+    private $_new_backend;
+
+    /** @var string */
+    private $_old_backend_type;
+
+    /** @var string */
+    private $_new_backend_type;
+
+    /** @var array */
+    private $_options = [];
+
+
+    /**
+    * Do stuff
+    **/
+    public function run(array $options)
+    {
+        ini_set('memory_limit', '1024M');
+
+        $this->_options = $options;
+        $this->_old_backend_type = $options['backend_source'];
+        $this->_new_backend_type = $options['backend_target'];
+
+        Worker::message("Migrating files to new backend: {$this->_new_backend_type}");
+
+        // If update is selected, we need to update the database records only
+        if ($this->_options['action'] == 'update') {
+
+            Worker::message("Updating database records only");
+            $this->updateDatabaseRecords();
+
+            Worker::message('');
+            Worker::success();
+        }
```

Comment:
This whole worker might read better if it were broken in a method per action to contain the gritty details. Halfway there already.

- updateDatabaseRecords()
- prepareFileCopy()

Then this core method breaks down into just the high level decisions to see what does/doesn't happen.

(worker logs omitted)

```php
switch ($action) {
   case 'prepare':
      $this->prepareFileCopy();
      break;
   case 'update':
      $this->updateDatabaseRecords();
      break;
   case 'full':
      $this->prepareFileCopy();
      $this->updateDatabaseRecords();
      break;
   default:
      throw new InvalidArgumentException("Unknown action: {$action}");
}
```

Reply:
That would read better, agreed!